### PR TITLE
[kubelet] Support dynamic tokens (Bound Service Account Token Volume)

### LINKED ISSUE #10604: Kubernetes api server metrics: reload bearer token to avoid failing on stale token
**Describe what happened:**
our logs show that dd agent uses an outdated service account token
```
{"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Metadata", ... "stage":"ResponseComplete","requestURI":"/metrics","verb":"get","user" ... "authentication.k8s.io/stale-token":"seconds after warning threshold: 3337", ...
```

**Describe what you expected:**
the agent should reload it's bearer token <every 10min or when it expires

**Steps to reproduce the issue:**
Deploy agent into a cluster that has BoundServiceAccountTokenVolume=true (new default in 1.22+) and make it scrape the kubernetes api server

**Additional environment details (Operating System, Cloud provider, etc):**
Agent/7.32.3

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Support refreshing the service account's auth token needed for [Bound Service Account Token Volume](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume) (k8s v1.22+)

### Motivation

fixes https://github.com/DataDog/datadog-agent/issues/10604

### Additional Notes

- This PR makes the kubelet python check supports dynamic auth tokens as well through rtloader: the Python check [polls the connection config](https://github.com/DataDog/integrations-core/blob/4a9d8fbacfddc90eb64c904b82351c2e033ead9a/kubelet/datadog_checks/kubelet/kubelet.py#L315) from Go on every check run (with 1 minute cache in the Go code now)
- This PR doesn't cover the `kube_apiserver_metrics` check: A separate fix needs to be done in `integrations-core` for openmetrics-based k8s integrations

### Describe how to test/QA your changes

On Kuberentes 1.22+, you can mount a projected volume with shorter token `expirationSeconds` parameter - Make sure the token gets updated in `/var/run/secrets/kubernetes.io/serviceaccount/token` regularily and the kubelet check and all components that rely on kubelet are working fine (tagger, workloadmeta, AD)

```
# in the pod spec
automountServiceAccountToken: false
```

```
volumeMounts:
  - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
    name: kube-api-access-<hash>
    readOnly: true
```

```
volumes:
- name: kube-api-access-<hash>
  projected:
    defaultMode: 420
    sources:
    - serviceAccountToken:
        expirationSeconds: 600
        path: token
    - configMap:
        items:
        - key: ca.crt
          path: ca.crt
        name: kube-root-ca.crt
    - downwardAPI:
        items:
        - fieldRef:
            apiVersion: v1
            fieldPath: metadata.namespace
          path: namespace
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/util/kubernetes/kubelet/kubelet_test.go [hunk 1]
```diff
@@ -597,15 +597,12 @@ func (suite *KubeletTestSuite) TestKubeletInitTokenHttps() {
 	<-k.Requests // Throwing away first GET
 
 	assert.Equal(suite.T(), fmt.Sprintf("https://127.0.0.1:%d", kubeletPort), ku.kubeletClient.kubeletURL)
-	assert.Equal(suite.T(), "bearer fakeBearerToken", ku.kubeletClient.headers.Get("Authorization"))
```
Comment (ahmed-mez) [inline pkg/util/kubernetes/kubelet/kubelet_test.go:600]: `transport.NewBearerAuthWithRefreshRoundTripper` handles the headers for us (asserted below)

File: pkg/util/kubernetes/kubelet/kubelet_test.go [hunk 2]
```diff
@@ -597,15 +597,12 @@ func (suite *KubeletTestSuite) TestKubeletInitTokenHttps() {
...
-	assert.True(suite.T(), ku.kubeletClient.client.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify)
```
Comment (ahmed-mez) [inline pkg/util/kubernetes/kubelet/kubelet_test.go:601]: `transport.NewBearerAuthWithRefreshRoundTripper` returns the `http.RoundTripper` interface, and the implementation is not public - it's impossible to cast here.
This assertion should be covered anyway by these lines below
```golang
	assert.Equal(suite.T(), "ok", string(b))
	assert.Equal(suite.T(), 200, code)
```

**Review by djmitche — APPROVED** [review]

**Review by cswatt — COMMENTED** [review]
no docs review required

File: pkg/collector/python/kubeutil.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ func GetKubeletConnectionInfo(payload **C.char) {
 		}
 
 		creds = string(data)
-		cache.Cache.Set(kubeletCacheKey, creds, 5*time.Minute)
+		cache.Cache.Set(kubeletCacheKey, creds, 1*time.Minute)
```
Comment (ahmed-mez) [inline pkg/collector/python/kubeutil.go:83]: see https://github.com/kubernetes/client-go/blob/28ccde769fc5519dd84e5512ebf303ac86ef9d7c/transport/token_source.go#L69-L82

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**